ement-room.el

```diff
@@ -2687,18 +2681,13 @@ Format defaults to `ement-room-message-format-spec', which see."
             (insert-and-inherit
              (propertize " "
                          'display `((margin right-margin) ,string))))))
-      (when-let* ((data (map-elt (ement-event-local event) (intern (concat "url-previews-"
-                                                                           (ement-event-id event)))))
-                  (num (or 1 (map-elt data
-                                      (intern (concat "url-previews-" (ement-event-id event)))))))
+      (when-let* ((data (ement-event-local event))
+                  (num (map-elt data 'url-preview-n)))
```

Comment:
Why store the number of previews?  They should probably just go in a list, and then...

Reply:
Yeah, that was mainly just for testing. I forgot to remove it :p

I'll probably just use dolist and use a variable as a counter or something to insert the correct order.

Reply:
This is because the number of actual needed URL previews doesn't necessarily correlate with the number of elements in the list. Some previews may have not been finished fetching yet, and sometimes homeservers seem to even send back duplicate previews if the connection is slow.

It isn't only used in the loop below either, also in other places to quickly get the number of expected previews, and also as a quick cheap check to see whether previews are present or not.